### DISCUSSION TITLE
Semantic-release in non-package repositories

### DISCUSSION THREAD

**Discussion by boy51** [semantic-release/semantic-release#2001] [General]
I've used semantic-release successfully for publishing packages to private registries, and I like it more than using `standard-version` which was my go-to before.

Now I'm thinking to move my other projects to this package. However, I'm not sure about the workflow when it comes to repositories that I'm not trying to publish to a registry, but rather deploy to production.

Currently I just do my deploys to EB and such in a pipeline, which simply takes my `dist` folder or whatever from build job to create a new version in EB with the latest tag.

So I see 2 ways to do this, I could just set `private` in `package.json` so that the package is not published. Then instead of my usual flow where i run `npm run release` myself, and merge to master, I would just merge to master and have a job before the deploy job to create the version.

Another way which I don't have much experience with would be to have semantic-release publish to a registry as intended, and the deploy based on the dist-tags from registry.

How do you guys implement this library in repositories that are not packages? Do you still manage as a package and deploy from a registry?

**Comment by travi** [comment]
does the following, from our F.A.Q. section of our docs, answer your question? https://semantic-release.gitbook.io/semantic-release/support/faq#can-i-use-semantic-release-to-publish-non-javascript-packages

**Reply by boy51** [reply]
Nope, I was talking about the possibility of using a package registry not to distribute a package, but to use as something like an extra step to deploy a project.

For now I'm not using a registry by setting `private: true` in package.json and just using semantic-release for versioning.

Thanks for this great library :)
